src/ucp/core/ucp_am.c

```diff
@@ -1101,6 +1120,12 @@ ucp_am_invoke_cb(ucp_worker_h worker, ucp_am_hdr_t *am_hdr,
                          flags);
 }
 
+static UCS_F_ALWAYS_INLINE int ucp_am_ep_is_valid(ucp_ep_h ep)
+{
+    return ((ep != NULL) &&
+            !(ep->flags & (UCP_EP_FLAG_CLOSED | UCP_EP_FLAG_FAILED)));
```

Comment:
maybe remove ep from ptr map before closed? avoid branch
@evgeny-leksikov

Reply:
do you mean directly from `ucp_ep_close_nb` and before request is completed?

Reply:
yes

Reply:
IIRC we don't have any EPs wire exchanges which can't be dropped after EP started closing, so should work

Reply:
should I do it here, or it is a separate PR?

Reply:
here